catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/WorkspaceTransformation.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer;
+
+import javax.annotation.Nullable;
+
+/**
+ * A representation of one key-value transformation for the {@link WorkspaceTransformer}.
+ * Implementations of this interface can be used to transform JSON-style key-value pairs into
+ * metacard attributes and vice versa, especially from and into workspace metacards.
+ *
+ * <p><b> This code is experimental. While this interface is functional and tested, it may change or
+ * be removed in a future version of the library. </b>
+ *
+ * @param <M> the expected type of metacard values handled by this {@link WorkspaceTransformation}
+ * @param <J> the expected type of JSON-style key-value pair values handled by this {@link
+ *     WorkspaceTransformation}
+ */
+public interface WorkspaceTransformation<M, J> {
```

Comment:
I think this interface might be simpler if we broke into two parts, one responsible for mapping metacards to json and another mapping json into metacards.

Reply:
I would disagree. I put them in the same interface because this transofmration is meant to be used in two directions; each change from JSON to metacard should have a corresponding change back from metacard to JSON and vice versa. They should be tightly coupled.